### PR TITLE
feat: ML docker image ControlNet support

### PR DESCRIPTION
**Pull request by SeanChao**

### CODE REVIEW

**Review by wifimug — APPROVED** [review]
LGTM

File: src/machine_learning/connector.py [hunk 1]
```diff
@@ -34,10 +35,10 @@ def generate_image(api_client: StableDiffusionAPI, text_prompt: str, negative_pr
 
     # Add a controlnet segmentation to the Txt2ImgConfig object
     if not disable_controlnet:
-        txt2img_config.add_controlnet_segmentation("control_v11p_sd15_seg [e1f51eb9]", read_segmentation_mask("test_seg1.png"))
+        txt2img_config.add_controlnet_segmentation("control_v11p_sd15_seg [e1f51eb9]", read_segmentation_mask("test_seg.png"))
 
     # Print the Txt2ImgConfig object as a dictionary
```
Comment (DanielSwarup) [inline src/machine_learning/connector.py:40]: I would recommend removing this print statement completely.

File: src/machine_learning/connector.py [hunk 2]
```diff
@@ -64,4 +65,11 @@ def generate_image(api_client: StableDiffusionAPI, text_prompt: str, negative_pr
     height = int(os.environ.get("HEIGHT", "512"))
 
     sd_api_client = StableDiffusionAPI(url)
-    generate_image(sd_api_client, text_prompt, disable_controlnet=disable_controlnet, width=width, height=height)
+    for _i in range(3):
```
Comment (DanielSwarup) [inline src/machine_learning/connector.py:68]: I would highly recommend against using time.sleep() instead, maybe listening to status updates and then triggering or using some non-blocking asynchronous methods

**Review by DanielSwarup — CHANGES_REQUESTED** [review]

File: src/machine_learning/connector.py (hunk 2 above)
Comment (SeanChao) [reply src/machine_learning/connector.py:68]: The API is external to the code. I can hardly discover any alternative approach to navigate around it.

**Review by DanielSwarup — APPROVED** [review]
LGTM!

**Pull request merged**